### ISSUE TITLE
Missing server callback on unauthorized

### ISSUE THREAD

**Issue by lerayne** [auth0/socketio-jwt#121]
Hello!

Nice component and article, though I haven't found any way to perform some actions on the server in case there is wrong token or no token. In other words, in my connection I want to give unauthorized user access to some part of data (read only). How can I do that? connection.on('unauthorized') doesn't work on server.

**Comment by jamespedid** [comment]
I agree with this as well. Some authentication requirements only dictate that a user elevate their permissions when viewing a certain area of the product.

**Comment by FaizanZahid** [comment]
facing same issue. i want to console.log when user is unauthorised.

**Comment by pat-flew** [comment]
There is a 'required' option which, if false, will prevent the server from disconnecting the socket if it does not auth within the timeout. You should be able to do something like

`server.on('connection', jwt.authorize({
    secret: JWT_SECRET,
    required: false
})).on('authenticated', socket => {});

server.on('connection', function(socket) {
 // socket.decoded_token should be available here once auth has happened
});`
